V allows a type to be used 2 times in a sum type declaration, ending in a C error

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relvant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** latest commit
**OS:** Linux Mint 20

**What did you do?**
I have made the declaration of a sumtype with the same type repeated 2 times, as seen here:
```v
struct Foo {}

type FooType = Foo | Foo

fn main() {
	_ := FooType{}
}
```

**What did you expect to see?**
An error message, for example:
```v
main.v:3:16: error: the type `Foo` is already defined in the sumtype `FooType`
    1 | struct Foo {}
    2 | 
    3 | type FooType = Foo | Foo
      |                      ~~~
    4 | 
    5 | fn main() {
```
 
**What did you see instead?**
A C compiler error:
```c
==================
/tmp/v/main.10378869689815410124.tmp.c: In function ‘v_typeof_sumtype_59’:
/tmp/v/main.10378869689815410124.tmp.c:1476:3: error: duplicate case value
 1476 |   case 58: return "Foo";
      |   ^~~~
/tmp/v/main.10378869689815410124.tmp.c:1475:3: note: previously used here
 1475 |   case 58: return "Foo";
      |   ^~~~
/tmp/v/main.10378869689815410124.tmp.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-unknown-warning-option’
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -345,7 +345,7 @@ pub fn (mut c Checker) type_decl(node ast.TypeDecl) {
 				}
 				mut sym := c.table.get_type_symbol(variant.typ)
 				if sym.name in found_sym_names {
-					c.error('sum type cannot hold the type `$sym.name` multiple times', node.pos)
+					c.error('sum type cannot hold the type `$sym.name` more than once', variant.pos)
 				} else if sym.kind == .placeholder {
 					c.error("type `$sym.name` doesn't exist", node.pos)
 				} else if sym.kind == .interface_ {

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -338,16 +338,20 @@ pub fn (mut c Checker) type_decl(node ast.TypeDecl) {
 		}
 		ast.SumTypeDecl {
 			c.check_valid_pascal_case(node.name, 'sum type', node.pos)
+			mut found_sym_names := []string{}
 			for variant in node.variants {
 				if variant.typ.is_ptr() {
 					c.error('sum type cannot hold a reference type', variant.pos)
 				}
 				mut sym := c.table.get_type_symbol(variant.typ)
-				if sym.kind == .placeholder {
+				if sym.name in found_sym_names {
+					c.error('sum type cannot hold the type `$sym.name` multiple times', node.pos)
+				} else if sym.kind == .placeholder {
 					c.error("type `$sym.name` doesn't exist", node.pos)
 				} else if sym.kind == .interface_ {
 					c.error('sum type cannot hold an interface', node.pos)
 				}
+				found_sym_names << sym.name
 			}
 		}
 	}

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -345,7 +345,8 @@ pub fn (mut c Checker) type_decl(node ast.TypeDecl) {
 				}
 				mut sym := c.table.get_type_symbol(variant.typ)
 				if sym.name in found_sym_names {
-					c.error('sum type cannot hold the type `$sym.name` more than once', variant.pos)
+					c.error('sum type cannot hold the type `$sym.name` more than once',
+						variant.pos)
 				} else if sym.kind == .placeholder {
 					c.error("type `$sym.name` doesn't exist", node.pos)
 				} else if sym.kind == .interface_ {

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -338,21 +338,21 @@ pub fn (mut c Checker) type_decl(node ast.TypeDecl) {
 		}
 		ast.SumTypeDecl {
 			c.check_valid_pascal_case(node.name, 'sum type', node.pos)
-			mut found_sym_names := []string{}
+			mut names_used := []string{}
 			for variant in node.variants {
 				if variant.typ.is_ptr() {
 					c.error('sum type cannot hold a reference type', variant.pos)
 				}
 				mut sym := c.table.get_type_symbol(variant.typ)
-				if sym.name in found_sym_names {
-					c.error('sum type cannot hold the type `$sym.name` more than once',
+				if sym.name in names_used {
+					c.error('sum type $node.name cannot hold the type `$sym.name` more than once',
 						variant.pos)
 				} else if sym.kind == .placeholder {
 					c.error("type `$sym.name` doesn't exist", node.pos)
 				} else if sym.kind == .interface_ {
 					c.error('sum type cannot hold an interface', node.pos)
 				}
-				found_sym_names << sym.name
+				names_used << sym.name
 			}
 		}
 	}

--- a/vlib/v/checker/tests/sum_type_multiple_type_define.out
+++ b/vlib/v/checker/tests/sum_type_multiple_type_define.out
@@ -1,4 +1,4 @@
-vlib/v/checker/tests/sum_type_multiple_type_define.vv:3:22: error: sum type cannot hold the type `Foo` more than once
+vlib/v/checker/tests/sum_type_multiple_type_define.vv:3:22: error: sum type FooType cannot hold the type `Foo` more than once
     1 | struct Foo {}
     2 |
     3 | type FooType = Foo | Foo

--- a/vlib/v/checker/tests/sum_type_multiple_type_define.out
+++ b/vlib/v/checker/tests/sum_type_multiple_type_define.out
@@ -1,0 +1,5 @@
+vlib/v/checker/tests/sum_type_multiple_type_define.vv:3:22: error: sum type cannot hold the type `Foo` more than once
+    1 | struct Foo {}
+    2 |
+    3 | type FooType = Foo | Foo
+      |                      ~~~

--- a/vlib/v/checker/tests/sum_type_multiple_type_define.vv
+++ b/vlib/v/checker/tests/sum_type_multiple_type_define.vv
@@ -1,0 +1,3 @@
+struct Foo {}
+
+type FooType = Foo | Foo

